Update reference documentation

cdrage opened the pull request:
This updates the reference documenation in order to have a clearer image
in terms of what's going on with OpenCompose.

We also increase the spacing in-between the titles and rename a few
specification names (ex. PortSpec to Port) for simplicity.

We also switch to example-based learning. Going through each part of the
specification full-example.

cdrage commented:
ping @pradeepto @surajssd @containscafeine

kadel commented on docs/file-reference.md:
```diff
@@ -1,8 +1,8 @@
-# OpenCompose file reference documentation
+## OpenCompose file reference
 
+_A full example of the OpenCompose specification:_
```
Not sure if we want to show *full* example of OpenCompose specification.
Really soon OpenCompose will have a lot of different features, only small number of complicated apps will use everything. 
A full example might give a wrong impression that OpenCompose is complicated.

kadel commented on docs/file-reference.md:
````diff
@@ -37,104 +40,135 @@ volumes:
+### Container
+
+```yaml
+services:
````
My original idea was to describe just the given section of the file without distracting it by other "upper"  sections.

Here that would be just what is inside the `<Container>`.

If you think that  was too confusing we can stick with that, I'm ok with that.

cdrage replied:
@kadel A bit confused, I believe that line 138 example (what you comment on here), goes over every possible value of <Container>, having the full example on the top gives users a quick reference / being able to quickly go over and iterate through each value.

Putting "-services" and "-name:foobar" shows WHERE `-container` goes.

kadel commented on docs/file-reference.md:
````diff
@@ -169,57 +203,69 @@ This overrides default Command from Docker container image.
+
+```yaml
+...
+env:
````
this should probably show everything from the top, so it's consistent with rest of the examples.

```yaml
services:
- name: foobar
  ...
  containers:
    ...
    env:
    - name: foo
      value: bar
    ...
  ...
```

cdrage replied:
Done.

kadel commented on docs/file-reference.md:
````diff
@@ -169,57 +203,69 @@ This overrides default Command from Docker container image.
+### Port
 
+```yaml
+services:
````
This should show that containers are in an array

``` yaml
services:
- name: foobar
  ...
  containers:
    - ...
      ports:
        - port: 8080:80
          type: internal
          host: domain.tld
          path: /admin
    ...
  ...
```

cdrage replied:
done.

kadel commented on docs/file-reference.md:
```diff
@@ -268,103 +318,125 @@ Path based ingresses specify a path component that can be compared against a URL
+services:
+- name: foobar
+  ...
+  containers:
```
Same as my comment in ports. It should show that containers are in an array

cdrage replied:
Done!

kadel commented on docs/file-reference.md:
```diff
@@ -268,103 +318,125 @@ Path based ingresses specify a path component that can be compared against a URL
 Name of the EmptyDir volume. 
 
 
 ## volumes
```
This is top level section. It should have same headline as `services` and `version` and you prepend others with "Section: ".
It should be same "Section: Volumes"

cdrage replied:
Ah, missed that. Thanks.

kadel reviewed (commented):
Thank you for cleaning up this a bit.
Just a few questions and pointers :innocent:

cdrage committed:
Update reference documentation

This updates the reference documenation in order to have a clearer image
in terms of what's going on with OpenCompose.

We also increase the spacing in-between the titles and rename a few
specification names (ex. PortSpec to Port) for simplicity.

We also switch to example-based learning. Going through each part of the
specification full-example.

cdrage commented:
Thanks for the reviews @kadel 

I've gone ahead and updated the PR.
